### PR TITLE
Minor FieldTemplate cleanup

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

These changes are because I was looking into ways of enforcing "proper" form configuration as part of https://github.com/department-of-veterans-affairs/va.gov-team/issues/10095. It's all very minor cleanup-type refactoring that I did so that what was happening would be more obvious.

## Testing done

Unit testing.

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms-system/src/js/components/FieldTemplate.jsx [hunk 1]
```diff
@@ -24,9 +31,6 @@ export default function FieldTemplate(props) {
     Object.keys(formContext.touched).some(touched => id.startsWith(touched));
   const hasErrors =
     (formContext.submitted || isTouched) && rawErrors && rawErrors.length;
-  const requiredSpan = required ? (
-    <span className="schemaform-required-span">(*Required)</span>
-  ) : null;
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/components/FieldTemplate.jsx:29]: I replaced this const with a functional component in an effort to reduce clutter within the `<FieldTemplate>` component and make the markup more declarative by using components.

File: src/platform/forms-system/src/js/components/FieldTemplate.jsx [hunk 2]
```diff
@@ -3,6 +3,13 @@ import _ from 'lodash/fp'; // eslint-disable-line no-restricted-imports
 import classNames from 'classnames';
 // import environment from 'platform/utilities/environment';
 
+// We may want this in its own file
+function RequiredSpan({ show }) {
+  return show ? (
+    <span className="schemaform-required-span">(*Required)</span>
+  ) : null;
+}
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/components/FieldTemplate.jsx:11]: The `show` prop will determine if the required span is rendered.

File: src/platform/forms-system/src/js/components/FieldTemplate.jsx [hunk 3]
```diff
@@ -36,28 +40,16 @@ export default function FieldTemplate(props) {
...
-  let errorClass;
-  if (hasErrors) {
-    errorClass = isDateField ? 'input-error-date' : 'usa-input-error';
-    errorSpanId = `${id}-error-message`;
-    errorSpan = (
-      <span className="usa-input-error-message" role="alert" id={errorSpanId}>
-        <span className="sr-only">Error</span> {rawErrors[0]}
-      </span>
-    );
-  }
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/components/FieldTemplate.jsx:56]: All of this was in-lined.

File: src/platform/forms-system/src/js/components/FieldTemplate.jsx [hunk 4]
```diff
@@ -108,7 +100,15 @@ export default function FieldTemplate(props) {
...
-      {errorSpan}
+      {hasErrors && (
+        <span
+          className="usa-input-error-message"
+          role="alert"
+          id={`${id}-error-message`}
+        >
+          <span className="sr-only">Error</span> {rawErrors[0]}
+        </span>
+      )}
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/components/FieldTemplate.jsx:111]: Here is the in-lining of `errorSpan` and associated variables.

File: src/platform/forms-system/src/js/components/FieldTemplate.jsx [hunk 5]
```diff
@@ -85,20 +77,20 @@ export default function FieldTemplate(props) {
...
+      <RequiredSpan show={required} />
     </label>
   );
 
   // Don't render hidden or empty labels - prevents duplicate IDs on review &
   // submit page
   const showLabel =
     !uiSchema['ui:options']?.hideLabelText &&
-    (typeof label !== 'string' || (requiredSpan || label.trim()));
+    (typeof label !== 'string' || (required || label.trim()));
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/components/FieldTemplate.jsx:93]: `requiredSpan` was truthy if `required` was truthy, and it makes more sense to check `required` here anyway.

**Review by rianfowler — APPROVED** [review]

**Comment by bkjohnson** [comment]
Closing this since refactoring the failing nightwatch tests this causes isn't a current priority.

**Pull request closed without merging**